src/main/app/src/Components/LandscapeComponent/Landscape/Landscape.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect, useContext, ReactElement } from 'react';
+
+import { ReactSvgPanZoomLoader, SvgLoaderSelectElement } from 'react-svg-pan-zoom-loader';
+import { ReactSVGPanZoom, TOOL_AUTO, Tool, Value } from 'react-svg-pan-zoom';
+import { useParams } from 'react-router-dom';
+
+import LandscapeItem from '../Item/LandscapeItem';
+import Command from '../../CommandComponent/Command';
+import GenericModal from '../../ModalComponent/GenericModal';
+import LandscapeContext from '../../../Context/Landscape.context';
+import CommandContext from '../../../Context/Command.context';
+
+import './Landscape.scss';
+import { ILandscape } from '../../../interfaces';
+
+const Landscape: React.FC = () => {
+  const [tool, setTool] = useState<Tool>(TOOL_AUTO);
+
+  // It wants a value or null but if we defined it as null it throws an error that shouldn't use null
+  // In their own documentation, they initialize it with {}, but that will invoke a typescript error
+  // @ts-ignore
+  const [value, setValue] = useState<Value>({});
+  const [modalContent, setModalContent] = useState<string | ReactElement | null>(null);
+  const [landscape, setLandscape] = useState<ILandscape | null>(null);
+  const [reloadLandscape, setReloadLandscape] = useState<boolean>(false);
+
+  const landscapeContext = useContext(LandscapeContext);
+  const commandContext = useContext(CommandContext);
+  const { identifier } = useParams();
+
+  useEffect(() => {
+    const index = landscapeContext.landscapes.findIndex(i => i.identifier === identifier);
+    setLandscape(landscapeContext.landscapes[index]);
+  }, [identifier, landscapeContext.landscapes, reloadLandscape]);
+
+  const reloadLandscapes = async () => {
```

Comment:
See my comment above, this is a case where the REST call logic could go into the context to make the Landscape component have less responsibilities. The LandscpaeContext would deliver the landscapes  ("getLandscapes" from above) and reload them (this reload) and you would only implement it once.

Reply:
yeah i wasnt happy with that either, didnt have much experience with context, i'll try it so thanks for the recommendation